Kernel/Syscalls/socket.cpp

```diff
@@ -227,18 +227,21 @@ ssize_t Process::sys$recvfrom(const Syscall::SC_recvfrom_params* user_params)
         return -EFAULT;
 
     int flags = params.flags;
-    sockaddr* addr = params.addr;
-    socklen_t* addr_length = params.addr_length;
+    Userspace<sockaddr*> user_addr = params.addr;
+    Userspace<socklen_t*> user_addr_length = params.addr_length;
 
     SmapDisabler disabler;
```

Comment:
Can we remove this now?

Reply:
I didn't look if the write to the data buffer is protected or not, that was the only thing that stopped me from removing it.